### PR TITLE
Remove saving instance state on variant list

### PR DESCRIPTION
**Pull request by timmutton**
Sometimes it was reinitialising the list wrong. Don't really need it anyway, getting the info from shared prefs isn't that expensive

@kamal-kamalmohamed

### CODE REVIEW

File: neanderthal/src/main/kotlin/au/com/outware/neanderthal/presentation/presenter/VariantListPresenter.kt [hunk 1]
```diff
@@ -24,35 +24,17 @@ class VariantListPresenter constructor(val view: ViewSurface,
...
         }
 
         adapter.add(variants)
     }
 
-    override fun onSaveInstanceState(outState: Bundle) {
-        if(variants.isNotEmpty()) {
-            outState.putInt(CURRENT_POSITION_KEY, currentPosition)
-            outState.putString(CURRENT_VARIANT_NAME_KEY, currentVariantName)
-            outState.putStringArrayList(VARIANTS_KEY, ArrayList<String>(variants))
```
Comment (kamal-kamalmohamed) [inline neanderthal/src/main/kotlin/au/com/outware/neanderthal/presentation/presenter/VariantListPresenter.kt:52]: Can we get rid of these constants now?

**Review by kamal-kamalmohamed — CHANGES_REQUESTED** [review]

File: neanderthal/src/main/kotlin/au/com/outware/neanderthal/presentation/view/activity/VariantListActivity.kt [hunk 2]
```diff
@@ -36,7 +36,7 @@ class VariantListActivity : AppCompatActivity(), VariantListPresenter.ViewSurfac
         adapter = VariantAdapter { name, position -> presenter.onItemSelected(name, position) }
 
         presenter = VariantListPresenter(this, adapter)
-        presenter.onCreate(savedInstanceState)
+        presenter.onCreate(null)
```
Comment (kamal-kamalmohamed) [inline neanderthal/src/main/kotlin/au/com/outware/neanderthal/presentation/view/activity/VariantListActivity.kt]: Why don't we pass the `Bundle` in anyway?

File: neanderthal/src/main/kotlin/au/com/outware/neanderthal/presentation/presenter/VariantListPresenter.kt (hunk 1 above)
Comment (timmutton) [reply neanderthal/src/main/kotlin/au/com/outware/neanderthal/presentation/presenter/VariantListPresenter.kt:52]: Good point

File: neanderthal/src/main/kotlin/au/com/outware/neanderthal/presentation/view/activity/VariantListActivity.kt (hunk 2 above)
Comment (timmutton) [reply neanderthal/src/main/kotlin/au/com/outware/neanderthal/presentation/view/activity/VariantListActivity.kt]: Fair question. updated the pattern a bit to throw in extras + saved instance state

**Review by kamal-kamalmohamed — APPROVED** [review]
Great!

**Pull request merged**